src/AutoRest.CSharp.V3/ClientModels/RequestBody.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using AutoRest.CSharp.V3.Pipeline;
+
+namespace AutoRest.CSharp.V3.ClientModels
+{
+    internal class RequestBody
+    {
+        public ConstantOrParameter Value { get; }
+        public SerializationFormat Format { get; }
+        public string Name { get; }
+        public ClientTypeReference Type { get; }
+
+        public RequestBody(ConstantOrParameter value, SerializationFormat format = SerializationFormat.Default)
+        {
+            Value = value;
+            Format = format;
+            Name = Value.IsConstant ? Value.Constant.ToValueString() : Value.Parameter.Name;
```

Comment:
`ToValueString()` is an evil method and would give a wrong result in a lot of cases. Something like `WriteConstant` should be used.

But in general we shouldn't be doing codegen on this level so I would drop name property and move this logic to the code generator itself.

Reply:
There is only 1 place right now where RequestBody was being written. And, it did both the name and type calculation right before it was used. So, I thought moving them here made sense. But, I can move name back there. I'd love to use `WriteConstant`. However, that will have to wait until we remove all the serialization writing out of Extensions.cs. That's where the name value feeds into right now. :-/

Reply:
Fixed.